### DISCUSSION TITLE
Will it support extensions?

### DISCUSSION THREAD

**Discussion by 3ri4nG0ld** [chase/awrit#33] [General]
I find it a very interesting and useful project, and I would like to know if there is any intention to support extensions.

In my daily life I have already replaced most of the tools I use with terminal tools, except the browser (and some others), and this project helps a lot. 
The only drawback is that I use some major extensions.

I understand that supporting extensions is not an easy thing to do, as it will probably involve adapting some things, but I just want to know if there is an intention to support them even if it is long term.

**Comment by chase** [comment]
Unfortunately, [Chromium Embedded Framework requires a custom implementation of all parts of the extension API](https://magpcss.org/ceforum/viewtopic.php?f=6&t=10242). This is non-trivial and would be a massive effort.

I wanted to integrate uBlock Origin using the proper extension API, but even the API surface area just to get that working was quite large:
https://github.com/chase/awrit/issues/3

I rarely have time to work on this project, so getting this setup would likely require a contribution from someone else.

**Comment by chase** [comment]
@3ri4nG0ld the latest version supports extensions :smiley:
